File: 401d10/challenges/fizz_buzz_tree/fizz_buzz_tree.py
from .. import bst


def fizz_buzz_tree(tree):
    """Takes a tree and changes out values if conditions are met."""
    def fizz_buzz(node):
        """."""
        if node.value is not int:
            return
        elif node.value % 3 == 0 and node.value % 5 == 0:
            node.value = 'fizzbuzz'
            return
        elif node.value % 3 == 0:
            node.vale = 'fizz'
            return
        elif node.value % 5 == 0:
            node.value = 'buzz'
            return
    tree.in_order()
    return tree


if __name__ == "__main__":
    fizz_buzz_tree([10, 18, 5, 2, 1, 15, 21])

File: 401d56/code-challenges/gradle/src/main/java/tree/FizzBuzzTree.java
package tree;

import java.util.ArrayList;

public class FizzBuzzTree {

    private Node root;

    class Node {
        Object data;
        Node left;
        Node right;

        Node(Object data) {
            this.data = data;
            right = null;
            left = null;
        }
    }

    ArrayList fizzBuzzTraversal(Node node, ArrayList newArray) {
        if (node != null) {
            fizzBuzzTraversal(node.left, newArray);
            int value = (int) node.data;
            if(value%3 == 0 && value%5 == 0){
                node.data = "fizzbuzz";
            }else if(value%3 == 0){
                node.data = "fizz";
            }else if (value%5 == 0){
                node.data = "buzz";
            }
            System.out.print(" " + node.data);
            newArray.add(node.data);
            fizzBuzzTraversal(node.right, newArray);
        }
        return newArray;
    }
}
